### ISSUE TITLE
pico_sntp_sync() attempts to perform DNS resolution on IP addresses

### ISSUE THREAD

**Issue by mateuszviste** [tass-belgium/picotcp#341]
When I call ```pico_sntp_sync("1.2.3.4", &cb_sntp)```, picoTCP tries to resolve "1.2.3.4" as a hostname, generating DNS requests. I do not believe this is right - rather ```pico_sntp_sync()``` should either accept an IP structure and let the application deal with DNS, or it should test the argument for being a valid IPv4 and/or IPv6 address before attempting any DNS resolution.
See tcpdump output below.

```
# tcpdump -ni tapdosemu port 123 or port 53
tcpdump: verbose output suppressed, use -v or -vv for full protocol decode
listening on tapdosemu, link-type EN10MB (Ethernet), capture size 262144 bytes
19:28:17.276176 IP 192.168.0.22.23756 > 192.168.0.1.53: 28637+ AAAA? 1.2.3.4. (30)
19:28:17.276261 IP 192.168.0.22.4310 > 192.168.0.1.53: 19183+ A? 1.2.3.4. (30)
19:28:17.276524 IP 192.168.0.1.53 > 192.168.0.22.23756: 28637 NXDomain 0/0/0 (30)
19:28:17.276669 IP 192.168.0.1.53 > 192.168.0.22.4310: 19183* 1/0/0 A 1.2.3.4 (46)
19:28:17.285585 IP 192.168.0.22.2553 > 1.2.3.4.123: NTPv4, Client, length 48
^C
5 packets captured
```

**Comment by fcapurso** [comment]
Accepted, thanks for reporting.

**Comment by frederikvs** [comment]
Seems to me that there are two ways of solving this : 
A) modify the current function to recognise an IP string, with all the unnecessary string processing involved.
B) add a new function to the API, which takes an IP address struct directly and uses it.

You may have noticed I have a slight preference for option B :-)

Probably not hard to do, but of course, these changes will have to be tested, and documented, as usual. 


Also, while glancing at the code I noticed that there is an ifdef for IPv6 support, but not for IPv4 support - it just assumes that IPv4 will be supported.
https://github.com/tass-belgium/picotcp/blob/master/modules/pico_sntp_client.c#L300 


@toonst do you feel like taking this up? Or shall we ask around in the community for somebody who wants to get started? Seems to me like this is a good issue to get somebody new involved in the code as well...

**Comment by toonst** [comment]
I only saw this now, sorry for not responding earlier.
I think a third option might be preferable:
remove the DNS resolving part altogether from the SNTP module, let `pico_sntp_sync` take an IP address en let the user resolve his hostname himself. This has the advantage that SNTP does not need DNS anymore, so both app layer modules are separated cleanly.

This is indeed a good issue for someone starting out, SNTP is a small module so quiet understandable. If someone is up for this please comment, otherwise I'm happy to start on this myself.

**Comment by frederikvs** [comment]
Downside of the third option is that it breaks an existing API. Though I agree it would be cleaner. 

Could we keep the original behaviour, but put it between extra `ifdef`s on DNS? That way, when both SNTP and DNS are enabled, existing projects can keep using this function, but if new code wants SNTP without DNS they can do it as well.

We'll also have to think about how to deprecate functions. If we decide the current function isn't the way to go, we need to be able to remove it in due time, but we can't just throw it out immediately and break stuff people might be using...

**Comment by toonst** [comment]
Good point. Yes I think that would be the best way to do it while keeping the API. We certainly have to think about a deprecation strategy. This probably also applies to #318 .
